src/DoocsSpectrum.cc

```diff
@@ -122,45 +87,28 @@ namespace ChimeraTK {
     D_spectrum::write(s);
   }
 
+  void DoocsSpectrum::addParameterAccessors() {
+    if(_startAccessor && _startAccessor->isReadable()) {
+      _doocsUpdater.addVariable(ChimeraTK::ScalarRegisterAccessor<float>(_startAccessor), _eqFct,
+          std::bind(&DoocsSpectrum::updateParameters, this));
+    }
+    if(_incrementAccessor && _incrementAccessor->isReadable()) {
+      _doocsUpdater.addVariable(ChimeraTK::ScalarRegisterAccessor<float>(_incrementAccessor), _eqFct,
+          std::bind(&DoocsSpectrum::updateParameters, this));
+    }
+  }
+
   void DoocsSpectrum::updateDoocsBuffer(const TransferElementID& transferElementId) {
     // Note: we already own the location lock by specification of the DoocsUpdater
 
-    // FIXME: A first  implementation is checking the data consistency here. Later this should be
-    // before calling this function because calling this function through a function pointer is
-    // comparatively expensive.
-    // Only check the consistency group if there is a macro pulse number associated.
     // There are only the processArray and the macro pulse number in the consistency
     // group. The limits are coming asynchronously and not for every macro pulse,
     // so we just take test latest we have.
-    // Do not check if update is coming from another DOOCS property mapped to the same variable (ID invalid), since
-    // the check would never pass. Such variables cannot use exact data matching anyway, since the update is triggered
-    // from the DOOCS write to the other property.
-    if(transferElementId.isValid() && _macroPulseNumberSource && !_consistencyGroup.update(transferElementId)) {
-      // data is not consistent (yet). Don't update the Doocs buffer.
-      // check if this will now throw away data and generate a warning
-      if(transferElementId == _processArray->getId()) {
-        if(!_doocsSuccessfullyUpdated) {
-          ++_nDataLossWarnings;
-          if(DoocsAdapter::checkPrintDataLossWarning(_nDataLossWarnings)) {
-            std::cout << "WARNING: Data loss in spectrum property " << _eqFct->name() << "/" << this->basename()
-                      << " due to failed data matching between value and macro pulse number (repeated "
-                      << _nDataLossWarnings << " times)." << std::endl;
-          }
-        }
-      }
-      _doocsSuccessfullyUpdated = false;
```

Comment:
You are not using _doocsSuccessfullyUpdated any more. What did it do? Why is it not needed any more?

Reply:
It is only used for the data loss warning, which is now move into `updateConsistency`.